gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTooFarFromTripShapeValidatorTest.java

```diff
@@ -0,0 +1,213 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.StopTooFarFromTripShapeNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsLocationType;
+import org.mobilitydata.gtfsvalidator.table.GtfsShape;
+import org.mobilitydata.gtfsvalidator.table.GtfsShapeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+
+@RunWith(JUnit4.class)
+public class StopTooFarFromTripShapeValidatorTest {
+    public static GtfsStop createStop(long csvRowNumber, String stopId, double stopLat,
+            double stopLon, GtfsLocationType locationType) {
+        return new GtfsStop.Builder()
+            .setCsvRowNumber(csvRowNumber)
+            .setStopId(stopId)
+            .setStopLat(stopLat)
+            .setStopLon(stopLon)
+            .setLocationType(locationType.getNumber())
+            .build();
+    }
+
+    public static GtfsStopTime createStopTime(
+            long csvRowNumber, String tripId, String stopId, int stopSequence) {
+        return new GtfsStopTime.Builder()
+            .setCsvRowNumber(csvRowNumber)
+            .setTripId(tripId)
+            .setStopId(stopId)
+            .setStopSequence(stopSequence)
+            .build();
+    }
+
+    public static GtfsShape createShape(long csvRowNumber, String shapeId, double shapePtLat,
+            double shapePtLon, int shapePtSequence) {
+        return new GtfsShape.Builder()
+            .setCsvRowNumber(csvRowNumber)
+            .setShapeId(shapeId)
+            .setShapePtLat(shapePtLat)
+            .setShapePtLon(shapePtLon)
+            .setShapePtSequence(shapePtSequence)
+            .build();
+    }
+
+    public static GtfsTrip createTrip(long csvRowNumber, String tripId, String shapeId) {
+        return new GtfsTrip.Builder()
+            .setCsvRowNumber(csvRowNumber)
+            .setTripId(tripId)
+            .setShapeId(shapeId)
+            .build();
+    }
```

Comment:
Why do we want to have this functions instead of directly calling the builder?

Reply:
Because if we directly call .build(), it will only set primary key or first key and sequence key, and no "conditionally required" fields are added. To meet our test requirements, some specific fields need to be called. That's why I have these methods.

Reply:
Fixed